src/videojs-media-sources.js

```diff
@@ -100,64 +98,156 @@
+          }
+
+          // Add the segments to the pendingBuffers array
+          self.pendingBuffers_.push(segment);
+        } else if (event.data.action === 'done') {
+          // All buffers should have been flushed from the muxer
+          // start processing anything we have received
+          self.processPendingSegments_();
         }
-        buffer.appendBuffer(event.data);
-      }.bind(this));
-
-      // aggregate buffer events
-      this.audioBuffer_.addEventListener('updatestart',
-                                         aggregateUpdateHandler(this, this.videoBuffer_, 'updatestart'));
-      this.videoBuffer_.addEventListener('updatestart',
-                                         aggregateUpdateHandler(this, this.audioBuffer_, 'updatestart'));
-      this.audioBuffer_.addEventListener('update',
-                                         aggregateUpdateHandler(this, this.videoBuffer_, 'update'));
-      this.videoBuffer_.addEventListener('update',
-                                         aggregateUpdateHandler(this, this.audioBuffer_, 'update'));
-      this.audioBuffer_.addEventListener('updateend',
-                                         aggregateUpdateHandler(this, this.videoBuffer_, 'updateend'));
-      this.videoBuffer_.addEventListener('updateend',
-                                         aggregateUpdateHandler(this, this.audioBuffer_, 'updateend'));
+      };
 
       // this buffer is "updating" if either of its native buffers are
       Object.defineProperty(this, 'updating', {
         get: function() {
-          return this.audioUpdating_ || this.videoUpdating_ ||
-            this.audioBuffer_.updating || this.videoBuffer_.updating;
+          return this.bufferUpdating_ ||
+            (this.audioBuffer_ && this.audioBuffer_.updating) ||
+            (this.videoBuffer_ && this.videoBuffer_.updating);
         }
       });
       // the buffered property is the intersection of the buffered
       // ranges of the native source buffers
       Object.defineProperty(this, 'buffered', {
         get: function() {
-          var start, end;
-          if (this.videoBuffer_.buffered.length === 0 ||
-              this.audioBuffer_.buffered.length === 0) {
+          var
+            start,
+            end,
+            starts = [],
+            ends = [],
+            lastIndex;
+
+          // Handle the case where there is no buffer
+          if ((this.videoBuffer_ && this.videoBuffer_.buffered.length === 0) ||
+              (this.audioBuffer_ && this.audioBuffer_.buffered.length === 0)) {
             return videojs.createTimeRange();
           }
-          start = Math.max(this.videoBuffer_.buffered.start(0),
-                           this.audioBuffer_.buffered.start(0));
-          end = Math.min(this.videoBuffer_.buffered.end(0),
-                         this.audioBuffer_.buffered.end(0));
+
+          // TODO: Calculate and return the true intersection of the buffered
+          //       property
```

Comment:
Well, for one, we only return a TimeRanges object with a single span even if more than one span exists. I'd like to think that for completeness we should consider returning a real object that is the intersection of both audio and video.

Although, if unification of the two streams is successful this whole "TODO" disappears. So.. yay!